File: lib/format/dedication.rb
require 'json'

module Format
  module Dedication

    def self.from_json(json_text)
      begin
        hash = JSON.parse(json_text)
      rescue
        return json_text
      end
      return "Donation made in #{hash['type'] || 'honor'} of #{hash['name']}. Note: #{hash['note']}"
    end
  end
end

File: javascripts/src/lib/dedication.ts
// License: LGPL-3.0-or-later
export interface Dedication {
  type?:'honor'|'memory',
  supporter_id?: number,
  name?:string
  contact?: {
    email?: string,
    phone?:string
    address?:string
  }
  note?:string
}

export function parseDedication(dedication?:string) : Dedication {
  if (!dedication || dedication == "")
    return {}
  return JSON.parse(dedication)
}

export function serializeDedication(dedication:Dedication) : string {
  return JSON.stringify(dedication)
}




